lib/omniauth/strategies/azure_activedirectory_v2.rb

```diff
@@ -113,6 +126,37 @@ def raw_info
 
         @raw_info
       end
+
+      # The below methods support the flow for using certificate-based client assertion authentication.
+      # See this documentation for details:
+      # https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-auth-code-flow#request-an-access-token-with-a-certificate-credential
+      def client_assertion_type
+        'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'
+      end
+
+      def client_assertion_claims(tenant_id, client_id)
+        {
+          'aud' => "https://login.microsoftonline.com/#{tenant_id}/oauth2/v2.0/token",
```

Comment:
It won't work in case of multitenant Microsoft Entra ID app . 
I case  tenant_id is not provided  'common' can be use